vite.config.js

```diff
@@ -1,68 +1,29 @@
- * see: https://github.com/hms-dbmi/viv/pull/469#issuecomment-877276110
- * @returns {import('vite').Plugin}
+ * @param {typeof pkg} pkg
+ * @param {{ forceExclude: Array<string> }} options
  */
-const bundleWebWorker = {
-    name: 'bundle-web-worker',
-    apply: 'serve', // plugin only applied with dev-server
-    async transform(_, id) {
-        if (/\?worker_file$/.test(id)) {
-            // just use esbuild to bundle the worker dependencies
-            const bundle = await esbuild.build({
-                entryPoints: [id],
-                inject: ['./src/alias/buffer-shim.js'],
-                format: 'iife',
-                bundle: true,
-                write: false
-            });
-            if (bundle.outputFiles.length !== 1) {
-                throw new Error('Worker must be a single module.');
-            }
-            return bundle.outputFiles[0].text;
-        }
-    }
-};
-
-// audit libraries to see if they use `Buffer` in a way that needs to be shimmed
+function determineExternalDependencies(pkg, options) {
+    // Packages that we have modified cannot be external because package consumers will not
+    // get our modifications. Ideally, we can avoid modifying these packages ourselves.
+    const patchedPackages = Object.keys(pkg.pnpm.patchedDependencies).map(
+        specifier => specifier.split('@').slice(0, specifier.startsWith('@') ? 2 : 1).join('@')
+    );
+    return [
+        ...Object.keys(pkg.dependencies),
+        ...Object.keys(pkg.peerDependencies)
+    ].filter(
+        dep => ![...patchedPackages, ...options.forceExclude].includes(dep)
+    );
 
-// We can't inject a global `Buffer` polyfill for the worker entrypoint using vite alone,
-// so we reuse the `bundle-web-worker` plugin to inject the buffer shim during production.
-const manualInlineWorker = {
-    apply: 'build',
-    async transform(code, id) {
-        if (
-            id.endsWith('bam-worker.ts?worker&inline') ||
-            id.endsWith('vcf-worker.ts?worker&inline') ||
-            id.endsWith('gff-worker.ts?worker&inline') ||
-            id.endsWith('bed-worker.ts?worker&inline')
-        ) {
-            const bundle = await bundleWebWorker.transform(code, id + '?worker_file');
-            const base64 = Buffer.from(bundle).toString('base64');
-            // https://github.com/vitejs/vite/blob/72cb33e947e7aa72d27ed0c5eacb2457d523dfbf/packages/vite/src/node/plugins/worker.ts#L78-L87
-            return `const encodedJs = "${base64}";
-const blob = typeof window !== "undefined" && window.Blob && new Blob([atob(encodedJs)], { type: "text/javascript;charset=utf-8" });
-export default function() {
-  const objURL = blob && (window.URL || window.webkitURL).createObjectURL(blob);
-  try {
-    return objURL ? new Worker(objURL) : new Worker("data:application/javascript;base64," + encodedJs, {type: "module"});
-  } finally {
-    objURL && (window.URL || window.webkitURL).revokeObjectURL(objURL);
-  }
-}`;
-        }
-    }
-};
```

Comment:
I'm suspect that this has not been working since upgrading vite previously. A better solution is to apply patches to dependencies specifically with pnpm.